### PR TITLE
Rewrite the buffer class, polish the CosineBasisLinear Module, change…

### PR DESCRIPTION
**Pull request by LiuYangMage**
… back StochasticActor

1, Rewrite the buffer class, use deque instead of list
2, Rename so that everthing is readable.
3, change back the StochasticActor

### CODE REVIEW

File: policies/gac/networks.py [hunk 1]
```diff
@@ -244,41 +253,58 @@ def _supervised_forward(self, state, taus, actions):
...
-        self.phi = CosineBasisLinear(n_basis_functions, cosine_hidden_size)
-        self.l2 = tf.keras.layers.Dense(200)
-        self.l3 = tf.keras.layers.Dense(action_dim)
+        self.module_type = 'StochasticActor'
+        self.noise_embed_dim = 400 // action_dim
+        self.action_dim = action_dim
+
+        self.state_embedding_layer = tf.keras.layers.Dense(
+            self.noise_embed_dim * self.action_dim,
+            activation=tf.keras.layers.LeakyReLU(alpha=0.01),
```
Comment (axlbrown) [inline policies/gac/networks.py:273]: why set alpha=0.01 instead of leave it as default?

Comment (LiuYangMage) [reply policies/gac/networks.py:273]: The default alpha in tf is 0.3, while in pytorch is 0.01. This is actually pretty big difference...My experience is that 0.01 works better, which is also consistent with their pytorch code.

Comment (axlbrown) [reply policies/gac/networks.py:273]: sounds good.

**Review by axlbrown — APPROVED** [review]
lgtm

**Pull request merged**